app.py

```diff
@@ -0,0 +1,61 @@
+from flask import Flask, render_template, redirect, url_for, request, session, flash
+from functools import wraps
+
+app = Flask(__name__)
+
+app.secret_key = "my precious" 
+
+def login_required(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        if 'logged_in' in session:
+            return f(*args, **kwargs)
+        else:
+            flash('You need to login first.')
```

Comment:
What does the `flash` function do?

Reply:
It displays the 'You need to login first' message to a user who isn't logged in in this case.

Reply:
👍